Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -578,7 +578,7 @@ async void DeviceInfoPropertyChanged(object sender, PropertyChangedEventArgs e)
 		void InsertPageBefore(Page page, Page before)
 		{
 			if (!_isAttachedToWindow)
-				PushCurrentPages();
+				PushCurrentPagesAsync();
```

Comment:
Should wait for the Task to complete, and depending on the returned value (bool), update or not the Toolbar?